### PR TITLE
12 add candidate starer group

### LINKED ISSUE #12: Migrate candidateStarter group

### PR DESCRIPTION
**Pull request by MaxTru**
closes #12 

Screenshot_20210712_160422

### CODE REVIEW

**Comment by MaxTru** [comment]
Adjusted reviewers due to vacation

File: src/provider/camunda-platform/properties/CandidateStarterProps.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+  const commandStack = useService('commandStack'),
+        translate = useService('translate'),
+        debounce = useService('debounceInput');
+
+  // Retrieve the process (via businessObject of bpmn:Process element or
+  // via processRef of bpmn:Participant)
+  const process = is(element, 'bpmn:Process') ?
+    getBusinessObject(element) :
+    getProcess(element);
```
Comment (pinussilvestrus) [inline src/provider/camunda-platform/properties/CandidateStarterProps.js:50]: Just a minimal thing, but we have the exact same code snippet in the `CandidateStarterUsers` property. Maybe we can have a helper for this.

File: src/provider/camunda-platform/properties/CandidateStarterProps.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        'camunda:candidateStarterGroups': value
+      }
+    });
+  };
+
+  return TextField({
+    element,
+    id: 'candidateStarterGroups',
+    label: translate('Candidate Starter Groups'),
+    description: translate('Specify more than one group as a comma separated list.'),
```
Comment (pinussilvestrus) [inline src/provider/camunda-platform/properties/CandidateStarterProps.js:70]: I know we have the descriptions in the old properties panel. However, we should, at some point, define a general help text/description strategy. When to add descriptions, and when not?

But, I'm okay with keeping with the existing implementation. 👍

**Review by pinussilvestrus — COMMENTED** [review]
Just one small finding. Otherwise, it looks good 👍 

One general remark: the description and label style currently don't differ too much in styling so it might be difficult to distinguish them (only by the order)

image

That's a general issue of `@bpmn-io/properties-panel` I think, and we should tackle it while doing the design polishing. /cc @andreasgeier

File: src/provider/camunda-platform/properties/CandidateStarterProps.js (hunk 1 above)
Comment (MaxTru) [reply src/provider/camunda-platform/properties/CandidateStarterProps.js]: :+1:

**Review by pinussilvestrus — APPROVED** [review]

**Comment by pinussilvestrus** [comment]
Let's fix the visual label/description distinction in the core. 👍

**Review by andreasgeier — COMMENTED** [review]
There are issues with the descriptions because the design finishing has not yet taken place.

- no distinction between labels and descriptions
- no clear alignment, it is hard to understand to which form element the text is related
- in general, we should not blow up the UI with permanent help texts => will be solved with the help text concept which provides togglable information

As a quick fix, I would suggest these CSS changes:

Screen Shot 2021-07-13 at 11 56 32

**Comment by andreasgeier** [comment]
> Considering we have this seperate issue, I would merge this one here, okay?

Sounds good. So I'm going to approve this issue without the styling changes.

**Review by andreasgeier — APPROVED** [review]
✅ &nbsp; Approved as discussed

**Pull request merged**